waiter/test/waiter/reporters_test.clj

```diff
@@ -0,0 +1,173 @@
+
+(deftest console-reporter-no-filter
+  (with-isolated-registry
+    (service-counter "service-id" "foo")
+    (service-counter "service-id" "foo" "bar")
+    (service-counter "service-id" "fee" "fie")
+    (counters/inc! (service-counter "service-id" "foo" "bar") 100)
+    (let [{:keys [ps out]} (make-printstream)
+          [console-reporter state] (r/make-console-reporter nil ps)]
+      (is (instance? ConsoleReporter console-reporter))
+      (.report console-reporter)
+      (is (= "
+-- Counters --------------------------------------------------------------------
+services.service-id.counters.fee.fie
+             count = 0
+services.service-id.counters.foo
+             count = 0
+services.service-id.counters.foo.bar
+             count = 100"
+             (->> (out)
+                  (clojure.string/split-lines)
+                  (drop 1)
+                  (clojure.string/join "\n"))))
+      (is (= {} @state)))))
+
+(deftest console-reporter-filter
+  (with-isolated-registry
+    (service-counter "service-id" "foo")
+    (service-counter "service-id" "foo" "bar")
+    (service-counter "service-id" "fee" "fie")
+    (counters/inc! (service-counter "service-id" "foo" "bar") 100)
+    (let [{:keys [ps out]} (make-printstream)
+          [console-reporter state] (r/make-console-reporter #"^.*fee.*" ps)]
+      (is (instance? ConsoleReporter console-reporter))
+      (.report console-reporter)
+      (is (= "
+-- Counters --------------------------------------------------------------------
+services.service-id.counters.fee.fie
+             count = 0"
+             (->> (out)
+                  (clojure.string/split-lines)
+                  (drop 1)
+                  (clojure.string/join "\n"))))
+      (is (= {} @state)))))
+
+(deftest graphite-reporter-no-filter
+  (with-isolated-registry
+    (service-counter "service-id" "foo")
+    (service-counter "service-id" "foo" "bar")
+    (service-counter "service-id" "fee" "fie")
+    (counters/inc! (service-counter "service-id" "foo" "bar") 100)
+    (let [actual-values (atom #{})
+          time (t/now)
+          graphite (reify GraphiteSender
+                     (flush [_])
+                     (getFailures [_] 0)
+                     (isConnected [_] true)
+                     (send [_ name value _] (swap! actual-values #(conj % (str name value)))))
+          [graphite-reporter state] (r/make-graphite-reporter nil "prefix" graphite)]
+      (is (instance? GraphiteReporter graphite-reporter))
+      (with-redefs [t/now (fn [] time)]
+        (.report graphite-reporter))
+      (is (= #{"prefix.services.service-id.counters.fee.fie.count0"
+               "prefix.services.service-id.counters.foo.count0"
+               "prefix.services.service-id.counters.foo.bar.count100"}
+             @actual-values))
+      (is (= {:last-reporting-time time
+              :failed-writes-to-server 0
+              :last-report-successful true} @state)))))
```

Comment:
Nitpick: lexicographic order

Reply:
Do we need to explicitly stop the graphite reporter?

Reply:
No, we didn't ever start it. We just called .report

Reply:
Okay.